weld/llvm.rs

```diff
@@ -1263,49 +1415,63 @@ impl LlvmGenerator {
                                 let (op_name, value) = try!(llvm_binop_vector(op, ty));
                                 let tmp = ctx.var_ids.next();
                                 let vec_prefix = format!("@{}", ll_ty.replace("%", ""));
-                                ctx.code
-                                    .add(format!("{} = call i32 {}.cmp({} {}, {} {})",
-                                                 tmp,
-                                                 vec_prefix,
-                                                 ll_ty,
-                                                 left_tmp,
-                                                 ll_ty,
-                                                 right_tmp));
-                                ctx.code
-                                    .add(format!("{} = icmp {} i32 {}, {}",
-                                                 bin_tmp,
-                                                 op_name,
-                                                 tmp,
-                                                 value));
-                                ctx.code
-                                    .add(format!("store {} {}, {}* {}",
-                                                 out_ty_str,
-                                                 bin_tmp,
-                                                 out_ty_str,
-                                                 llvm_symbol(output)));
+                                ctx.code.add(format!(
+                                    "{} = call i32 {}.cmp({} {}, {} {})",
+                                    tmp,
+                                    vec_prefix,
+                                    ll_ty,
+                                    left_tmp,
+                                    ll_ty,
+                                    right_tmp
+                                ));
+                                ctx.code.add(format!(
+                                    "{} = icmp {} i32 {}, {}",
+                                    bin_tmp,
+                                    op_name,
+                                    tmp,
+                                    value
+                                ));
+                                ctx.code.add(format!(
+                                    "store {} {}, {}* {}",
+                                    out_ty_str,
+                                    bin_tmp,
+                                    out_ty_str,
+                                    llvm_symbol(output)
+                                ));
                             }
                             _ => weld_err!("Illegal type {} in BinOp", print_type(ty))?,
                         }
                     }
+                    UnaryOp {
+                        ref output,
+                        op,
+                        ref child,
+                    } => {
+                        let op_name = try!(llvm_unaryop(op));
```

Comment:
See comment above; I think it makes more sense to just merge the functionality of these.